lib/fastlane/plugin/firebase_app_distribution/client/firebase_app_distribution_api_client.rb

```diff
@@ -123,57 +134,50 @@ def upload_binary(app_id, binary_path, platform)
       # Takes at least POLLING_INTERVAL_SECONDS between polling get_upload_status
       #
       # args
-      #   project_number - Firebase project number
       #   app_id - Firebase app ID
       #   binary_path - Absolute path to your app's aab/apk/ipa file
       #
       # Returns the release_id of the uploaded release.
       #
       # Crashes if the number of polling retries exceeds MAX_POLLING_RETRIES or if the binary cannot
       # be uploaded.
-      def upload(project_number, app_id, binary_path, platform)
+      def upload(app_id, binary_path, platform)
         binary_type = binary_type_from_path(binary_path)
 
-        upload_token = get_upload_token(project_number, app_id, binary_path)
-        upload_status_response = get_upload_status(app_id, upload_token)
-        if upload_status_response.success? || upload_status_response.already_uploaded?
-          UI.success("✅ This #{binary_type} has been uploaded before. Skipping upload step.")
-        else
-          unless upload_status_response.in_progress?
-            UI.message("⌛ Uploading the #{binary_type}.")
-            upload_binary(app_id, binary_path, platform)
-          end
-          MAX_POLLING_RETRIES.times do
-            upload_status_response = get_upload_status(app_id, upload_token)
-            if upload_status_response.success? || upload_status_response.already_uploaded?
-              UI.success("✅ Uploaded the #{binary_type}.")
-              break
-            elsif upload_status_response.in_progress?
-              sleep(POLLING_INTERVAL_SECONDS)
+        UI.message("⌛ Uploading the #{binary_type}.")
+        operation_name = upload_binary(app_id, binary_path, platform)
+
+        upload_status_response = get_upload_status(operation_name)
+        MAX_POLLING_RETRIES.times do
+          if upload_status_response.success? || upload_status_response.already_uploaded?
+            UI.success("✅ Uploaded the #{binary_type}.")
```

Comment:
Ooooooh could we update this message to specify where the binary resulted in a new release or if it was grouped in an existing release?

Reply:
Great call! Added two more cases for `RELEASE_UNMODIFIED` and `RELEASE_UPDATED`.

Reply:
Maybe change this message to explicitly point out that a new release was created?

Also: How about including the `display_version` and `build_version` in all three of these messages?

Reply:
Done.